### PR TITLE
fix/test: swap over contiguously initialized ticks with spacing of one

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #5558, #5571, 5574

## What is the purpose of the change

Adding swap test vectors for out given in where contiguous ticks are initialized with spacing of one. Both zero for one and one for zero are covered

This test helped uncover 2 problems that were separated into the following PRs:
- https://github.com/osmosis-labs/osmosis/pull/5581
- https://github.com/osmosis-labs/osmosis/pull/5578

This PR also addresses removing the hack described here: https://github.com/osmosis-labs/osmosis/issues/5571

It also addresses https://github.com/osmosis-labs/osmosis/issues/5574 with investigations done in `TestSwapOutGivenIn_PriceLimit_TickCross_ZeroForOne` to swap with sqrt price limit at tick boundaries and asserting that ticks are crossed as expected. I was unable to construct a test case that would reproduce the behavior defined in #5574

## Testing and Verifying

This change adds a test.

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/concentrated-liquidity/swaps_tick_cross_test.go [hunk 1]
```diff
@@ -878,6 +880,199 @@ func (s *SwapTickCrossTestSuite) TestSwapOutGivenIn_Tick_Initialization_And_Cros
...
+				nextTickToReachInCompute = nextTickToReachInCompute - 1
+			} else {
+				nextTickToReachInCompute = nextTickToReachInCompute + 1
+			}
+
+			return nextTickToReachInCompute, sdk.NewDecWithPrec(5, 1)
+		}
+
+		return expectedSwapEndTick, sdk.OneDec()
+	}
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/swaps_tick_cross_test.go:954]: Can you explain a little more about the multiplier here? I later realized we used 50 percent to stay within the same tick, but it was not clear when initially reading the code here.

Comment (p0mvn) [reply x/concentrated-liquidity/swaps_tick_cross_test.go:954]: You are correct. We use a 50% discount to stay within the same bucket. 50% picked arbitrarily. If we don't discount, the next tick will be crossed as we are running the token in estimation logic between ticks. For some test cases, we specifically want to test staying within the same bucket.

Let me know if there is anything else that I should elaborate on

Comment (czarcas7ic) [reply x/concentrated-liquidity/swaps_tick_cross_test.go:954]: Can you add that the 50 percent is arbitrary in the comments to make it easier for other reviewers?

Comment (p0mvn) [reply x/concentrated-liquidity/swaps_tick_cross_test.go]: done

File: x/concentrated-liquidity/swaps_tick_cross_test.go [hunk 2]
```diff
@@ -878,6 +880,200 @@ func (s *SwapTickCrossTestSuite) TestSwapOutGivenIn_Tick_Initialization_And_Cros
...
+				expectedSwapEndTicks[i] = types.MaxTick
+			} else {
+				expectedSwapEndTicks[i] = originalCurrentTick + swapTicksAway
+			}
+		}
+
+		return expectedSwapEndTicks
+	}
+
+	// computeExpectedValuesForTestOneForZero returns the tick to swap to during estimate computation and amountIn multiplier.
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/swaps_tick_cross_test.go:936]:
```suggestion
	// computeNextTickToReachAndMultiplier returns the tick to swap to during estimate computation and amountIn multiplier.
```

**Review by czarcas7ic — APPROVED** [review]
While the I don't have the most in depth understanding of the tick issues we have been facing, this PR logically follows for me

**Pull request merged**